velox/exec/FilterProject.cpp

```diff
@@ -53,7 +53,13 @@ FilterProject::FilterProject(
           "FilterProject"),
       hasFilter_(filter != nullptr),
       project_(project),
-      filter_(filter) {}
+      filter_(filter),
+      mergingVectorOutput_(std::make_shared<MergingVectorOutput>(
+          operatorCtx_->pool(),
+          driverCtx->queryConfig().preferredOutputBatchBytes(),
+          driverCtx->queryConfig().preferredOutputBatchRows(),
+          driverCtx->queryConfig().minMergingVectorOutputBatchBytes(),
+          driverCtx->queryConfig().minMergingVectorOutputBatchRows())) {}
```

Comment:
Set to 0 if there is no filter

Reply:
It is difficult for us to know whether the FilterProject will filter. However, even there is no filter, setting a value greater than 0 will have little impact on performance: Without filtering, the size of the FilterProject's output vector will be larger, and larger vectors than Min_merging_vector_output_batch_rows will not be merged .

Reply:
We want it to be merged in upstream operator instead of here if there is no filter

Reply:
@Yuhta by 'no filter', you mean `FilterProject::hasFilter == false`, right?

Reply:
Yes

Reply:
done